### PR TITLE
[INFINITY-1369] get diagnostics on test failure

### PR DESCRIPTION
**Pull request by jsrodman**
When --keep-workdir  is used; fetch dcos diagnostics if the test run fails.

### CODE REVIEW

**Comment by benclarkwood** [comment]
This all makes sense to me. I will defer to @mrbrowning for any Python concerns.

File: tools/clustinfo.py [hunk 1]
```diff
@@ -67,31 +84,59 @@ def stop_cluster(cluster):
...
 
+def fetch_diagnostics(tgt_dir):
+    "Collect diagnostic informaion from all clusters"
+    logger.info("Collecting diagnostics from clusters")
+    for cluster in _clusters:
+        cluster.start_diagnostics()
+        logger.info("Spawned diagnostics on cluster: %s", cluster.url)
+
+    for cluster in _clusters:
+        while True:
```
Comment (mrbrowning) [inline tools/clustinfo.py]: Could be expressed as `while cluster.diagnostics_complete()`, no?

File: tools/clustinfo.py [hunk 2]
```diff
@@ -105,6 +150,81 @@ def in_use(self):
...
+        os.makedirs(self.config_dir)
+        self.custom_env['DCOS_CONFIG'] = config_path
+
+        self._dcoscli_config_set('core.dcos_url', self.url)
+        self._dcoscli_config_set('core.ssl_verify', 'false')
+        self._dcoscli_config_set('core.reporting', 'True') # What does this do?
+        self._dcoscli_config_set('core.timeout', '10')
+
+        self._dcoscli_config_set('core.dcos_acs_token', self.auth_token)
+
```
Comment (mrbrowning) [inline tools/clustinfo.py]: Unneeded double newline here.

File: tools/clustinfo.py (hunk 1 above)
Comment (jsrodman) [inline tools/clustinfo.py]: Fair.

File: tools/clustinfo.py (hunk 2 above)
Comment (jsrodman) [inline tools/clustinfo.py]: Ok.

**Comment by jsrodman** [comment]
retest this please

**Comment by jsrodman** [comment]
retest this please

**Comment by jsrodman** [comment]
Due to hangs I've experienced with these commands outside CI, I'm tabling this approach.

**Pull request closed without merging**